src/tests/system/tests/test_passkey.py

```diff
@@ -241,3 +241,188 @@ def test_passkey__su_srv_not_resolvable(
+@pytest.mark.builtwith(client="passkey", provider="passkey")
+def test_passkey__su_multi_keys_for_same_user(
+    client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str
+):
+    """
+    :title: Check su authentication of user when multiple keys added for same user with
+            LDAP, IPA, AD and Samba server.
+    :setup:
+        1. Add a user with multiple mappings of passkey in LDAP, IPA, AD and Samba with passkey_mapping.
+        2. Setup SSSD client with FIDO and umockdev, start SSSD service.
+    :steps:
+        1. Check su authentication of the user.
+    :expectedresults:
+        1. User su authenticates successfully.
+    :customerscenario: False
+    """
+    suffix = type(provider).__name__.lower()
+    user_add = provider.user("user1").add()
+
+    if suffix == "ldap":
+        client.sssd.domain["local_auth_policy"] = "only"
+        for n in range(1, 5):
+            if bool(user_add.get(attrs=["passkey"])):
+                with open(f"{testdatadir}/passkey-mapping.{suffix}{n}") as f:
+                    attrs = {"passkey": f.read().strip()}
+                    user_add._modify(add=attrs)
+            else:
+                with open(f"{testdatadir}/passkey-mapping.{suffix}{n}") as f:
+                    user_add.passkey_add(f.read().strip())
+    else:
+        for n in range(1, 5):
+            with open(f"{testdatadir}/passkey-mapping.{suffix}{n}") as f:
+                user_add.passkey_add(f.read().strip())
+
+    client.sssd.start()
+
+    assert client.auth.su.passkey(
+        username="user1",
+        pin=123456,
+        device=f"{moduledatadir}/umockdev.device",
+        ioctl=f"{moduledatadir}/umockdev.ioctl",
+        script=f"{testdatadir}/umockdev.script.{suffix}",
+    )
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+@pytest.mark.builtwith(client="passkey", provider="passkey")
+def test_passkey__su_same_key_for_multi_user(
+    client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str
+):
+    """
+    :title: Check su authentication of user when same key added for multiplr user with LDAP, IPA, AD and Samba server.
+    :setup:
+        1. Add three users with same passkey mapping in LDAP, IPA, AD and Samba with passkey_mapping.
+        2. Setup SSSD client with FIDO and umockdev, start SSSD service.
+    :steps:
+        1. Check su authentication of the user1, user2 and user3.
+    :expectedresults:
+        1. User1, user2 and user3 su authenticates successfully with same mapping.
+    :customerscenario: False
+    """
+    suffix = type(provider).__name__.lower()
+
+    if suffix == "ldap":
+        client.sssd.domain["local_auth_policy"] = "only"
+
+    for user in ["user1", "user2", "user3"]:
+        user_add = provider.user(user).add()
+        with open(f"{testdatadir}/passkey-mapping.{suffix}") as f:
+            user_add.passkey_add(f.read().strip())
+
+        client.sssd.start()
```

Comment:
It is not necessary to start the SSSD service each time, but only once.

Reply:
updated